### PR TITLE
added solarEnvelopeAdvanced component

### PR DESCRIPTION
**Pull request by boris-p**

### CODE REVIEW

File: src/Ladybug_SolarEnvelopeAdvanced.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        self.finalPoint = Rhino.Geometry.Point3d(self.point[0],self.point[1],self.height) 
+        return self.finalPoint
+#class to organize all the data in a single sun object
+#properties for later use, when we'll get more comprehensive data from and epw file - hour, day, month, temperature, radiation
+class SingleSun:
+    def __init__(self, _alltitude, _azimuth):
+        self.alltitude = _alltitude
+        self.azimuth = _azimuth
+
+if _runIt:
```
Comment (mostaphaRoudsari) [inline src/Ladybug_SolarEnvelopeAdvanced.py]: @boris-p here you probably want to make sure that all the other required inputs are also provided. Something like:
```python
if _runIt and _baseSrf and len(_sunVectors)!=0 and (other required inputs):
```

File: src/Ladybug_SolarEnvelopeAdvanced.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+# under a Creative Commons Attribution-ShareAlike 3.0 Unported License.
+"""
+Use this component to generate a solar envelope for a given test surface, set of solar vectors, and context geometry that you want to ensure solar access to.  Solar envelopes are typically used to illustrate the volume that can be built within in order to ensure that a new development does not shade the surrounding properties for a given set of sun vectors.
+
+-
+Provided by Ladybug 0.0.59
+    
+    Args:
+        _baseSrf: A surface representing the area for which you want to create the solar envelope.
+        _obstacleCurves: A list of curves indicating the bottom borders of your surroundings for which you would like solar access to be kept.
```
Comment (mostaphaRoudsari) [inline src/Ladybug_SolarEnvelopeAdvanced.py]: is _obstacleCurves a required input? I assume user should be able to run the analysis also for a surface with no context? If it is an optional input then you need to move the _ to the end > obstacleCurves_

File: src/Ladybug_SolarEnvelopeAdvanced.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    Returns:
+        readMe!:Log of the component
+        finalPointList: A list of points representing the heights to which you can build without shading any of the _obstacleCurves from the input _sunVectors.
+        total_ms: The time that it took this component to run
+"""
+ghenv.Component.Name = 'Ladybug_SolarEnvelopeAdvanced'
+ghenv.Component.NickName = 'SolarEnvelopeAdvanced'
+ghenv.Component.Message = 'VER 0.0.59\nAPR_20_2015'
+ghenv.Component.Category = "Ladybug"
+ghenv.Component.SubCategory = "3 | EnvironmentalAnalysis"
```
Comment (mostaphaRoudsari) [inline src/Ladybug_SolarEnvelopeAdvanced.py]: I tend to ask everyone to put the components under WIP to be tested first and once we are sure about the performance we can move it to the main subcategory. For ladybug it is: "6 | WIP"

**Comment by mostaphaRoudsari** [comment]
@boris-p great work! I put some comments for a couple of minor changes, otherwise it looks great.

File: src/Ladybug_SolarEnvelopeAdvanced.py (hunk 3 above)
Comment (boris-p) [inline src/Ladybug_SolarEnvelopeAdvanced.py]: done

File: src/Ladybug_SolarEnvelopeAdvanced.py (hunk 2 above)
Comment (boris-p) [inline src/Ladybug_SolarEnvelopeAdvanced.py]: Thought that people could plug in the base surface for the obstacle curves, but this is simpler and more intuitive, done

File: src/Ladybug_SolarEnvelopeAdvanced.py (hunk 1 above)
Comment (boris-p) [inline src/Ladybug_SolarEnvelopeAdvanced.py]: done

**Comment by boris-p** [comment]
Updated the component. Let me know if you see anything else worth changing as this stage

**Comment by mostaphaRoudsari** [comment]
@boris-p let me know when you have tutorial uploaded and we can post it on Facebook so you get more people to test it.

**Comment by chriswmackey** [comment]
@boris-p and @mostaphaRoudsari ,
Sorry for being AWOL on this and glad that everything was integrated.  Looking forward to playing around with this once I get past my thesis.
-Chris

**Pull request merged**